recovery/cmd/backup_size.go

```diff
@@ -0,0 +1,188 @@
+package cmd
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	clusterPath         = "/var/lib/etcd/member/snap/db"
+	staticPodsPath      = "/etc/kubernetes/static-pod-resources/"
+	usrLocalPath        = "/usr/local"
+	kubeletPath         = "/var/lib/kubelet"
+	etcPath             = "/etc"
+	binPath             = "/etc/kubernetes/static-pod-resources/bin/"
+	backupDir           = "/var/recovery"
+	backupSizeSafetyNet = 10.00 // in GiB
+	units               = []string{"B", "KiB", "MiB", "GiB", "TiB", "EiB", "ZiB"}
+)
+
+type resource struct {
+	dirPath string
+}
+
+type resourceList struct {
+	resources *[]resource
+}
+
+// compareBackupToDisk verifies disk space required against available
+// disk space and returns boolean value
+func compareBackupToDisk() (bool, error) {
+
+	estimated, err := estimateDiskSize()
+	if err != nil {
+		log.Errorf("Couldn't calculate estimated disk space")
+		return false, err
+	}
+
+	freeDisk, err := diskPartitionSize()
+	if err != nil {
+		log.Errorf("Couldn't calculate free disk space")
+		return false, err
+	}
+
+	estimated, ebytes := sizeConversion(estimated)
+	freeDisk, fbytes := sizeConversion(freeDisk)
+
+	log.Infof("Available disk space : %.2f %s; Estimated disk space required for backup: %.2f %s \n", freeDisk, fbytes, estimated, ebytes)
+
+	// find the index of estimated and freedisk space in units[]
+	var ebytesID, fbytesID int
+	for i := range units {
+		if units[i] == ebytes {
+			ebytesID = i
+		}
+		if units[i] == fbytes {
+			fbytesID = i
+		}
+	}
+
+	if ebytesID <= fbytesID {
+		x := float64(fbytesID - ebytesID)
+		if x != 0.0 {
+			estimated /= 1024 * x
+		}
+		if freeDisk > backupSizeSafetyNet+estimated {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// estimateDiskSize calculate the required backup size
+// returns: disk size(int), error
+func estimateDiskSize() (float64, error) {
```

Comment:
Suggest renaming to something like estimateFsSpaceRequirements. The current function name sounds like it's getting the size of the disk.